### PR TITLE
throw error if the videoID returned is different

### PR DESCRIPTION
**Pull request by unixfox**
Closes https://github.com/iv-org/invidious/issues/3257

### CODE REVIEW

File: src/invidious/videos.cr [hunk 1]
```diff
@@ -909,6 +909,10 @@ def extract_video_info(video_id : String, proxy_region : String? = nil, context_
         "reason" => JSON::Any.new(reason),
       }
     end
+  elsif video_id != player_response.dig("videoDetails", "videoId")
+    # YouTube may return a different video player response than expected.
+    # See: https://github.com/TeamNewPipe/NewPipe/issues/8713
+    raise BrokenTubeException.new("Different video returned by YouTube compared to the one requested. Can't playback the video.")
```
Comment (SamantazFox) [inline src/invidious/videos.cr:915]:
```suggestion
    raise BrokenTubeException.new("YouTube returned the wrong video")
```

**Review by SamantazFox — COMMENTED** [review]
[Don't mind my previous (now deleted) comment]

We should probably put something in info["error"] instead of rising an exception. I'm pretty sure that rising an exception for the wrong video ID will break sonner or later.

Also, if we raise something, We should raise an `InfoException` (no trace) or just an `Exception`. A `BrokenTubeException` is used to give the name of a JSON object that we can't find.

EDIT: We can also create a dedicated `VideoNotAvailableException`

**Comment by unixfox** [comment]
> We should probably put something in info["error"] instead of rising an exception. I'm pretty sure that rising an exception for the wrong video ID will break sonner or later.

Then the user wouldn't have direct easy access to reporting the issue. Returning an info[error] means only a single message is displayed.

> Also, if we raise something, We should raise an `InfoException` (no trace) or just an `Exception`. A `BrokenTubeException` is used to give the name of a JSON object that we can't find.
> 
> EDIT: We can also create a dedicated `VideoNotAvailableException`

Ok for a  `VideoNotAvailableException`.

**Comment by SamantazFox** [comment]
> Ok for a `VideoNotAvailableException`.

Go for it, then!

**Comment by unixfox** [comment]
By throwing an `InfoException` I don't get the full error template, that's what I wanted to get...

**Comment by unixfox** [comment]
Ok I got it, please review again.

File: src/invidious/videos.cr [hunk 2]
```diff
@@ -909,6 +909,10 @@ def extract_video_info(video_id : String, proxy_region : String? = nil, context_
         "reason" => JSON::Any.new(reason),
       }
     end
+  elsif video_id != player_response.dig("videoDetails", "videoId")
+    # YouTube may return a different video player response than expected.
+    # See: https://github.com/TeamNewPipe/NewPipe/issues/8713
+    raise VideoNotAvailableException.new("Different video returned by YouTube compared to the one requested. Can't playback the video.")
```
Comment (SamantazFox) [inline src/invidious/videos.cr:915]: I'd reword the first sentence as follows:
"The video returned by Youtube isn't the requested one"

We can also remove the second sentence, as it will be obvious that the video can't be played (the user is on the error page).

Comment (unixfox) [reply src/invidious/videos.cr]: Reworded

**Review by SamantazFox — APPROVED** [review]
